src/components/HighlightedText/index.d.ts

```diff
@@ -0,0 +1,10 @@
+import { ComponentType } from 'react';
+import { BaseProps } from '../types';
+
+export interface HighlightedText extends BaseProps {
+    HitComponent?: ComponentType;
+    TextComponent?: ComponentType;
```

Comment:
ComponentType is a generic type and needs an argument of props we can pass something like:
```hitComponent?: ComponentType<{ children?: string }>;```